### PR TITLE
Expose CDI log verbosity to HCO CR

### PR DESCRIPTION
**Pull request by alromeros**
**What this PR does / why we need it**:

This Pull Request aims to add a new field in the HCO CR to expose CDI log verbosity. This will allow users who deploy CDI with HCO to tweak log verbosity.

**Reviewer Checklist**

> Reviewers are supposed to review the PR for every aspect below one by one. To check an item means the PR is either "OK" or "Not Applicable" in terms of that item. All items are supposed to be checked before merging a PR. 

- [ ] PR Message
- [ ] Commit Messages
- [ ] How to test
- [ ] Unit Tests
- [ ] Functional Tests
- [ ] User Documentation
- [ ] Developer Documentation
- [ ] Upgrade Scenario
- [ ] Uninstallation Scenario
- [ ] Backward Compatibility
- [ ] Troubleshooting Friendly

**Jira Ticket**:

```jira-ticket
NONE
```

**Release note**:

```release-note
Expose CDI's log verbosity to HCO CR
```

### CODE REVIEW

**Comment by alromeros** [comment]
Continuing with this after the CDI bump.

File: api/v1beta1/hyperconverged_types.go [hunk 1]
```diff
@@ -661,6 +661,8 @@ type Version struct {
 type LogVerbosityConfiguration struct {
 	// +optional
 	Kubevirt *v1.LogVerbosity `json:"kubevirt,omitempty"`
+	// +optional
```
Comment (nunnatsa) [inline api/v1beta1/hyperconverged_types.go:664]: Please add a comment with the field description.

File: controllers/operands/cdi_test.go [hunk 2]
```diff
@@ -515,6 +515,31 @@ var _ = Describe("CDI Operand", func() {
...
+				logVerbosity := int32(4)
+				hco.Spec.LogVerbosityConfig = &hcov1beta1.LogVerbosityConfiguration{CDI: &logVerbosity}
+				cdi, err := NewCDI(hco)
+
+				Expect(err).ShouldNot(HaveOccurred())
+				Expect(cdi).ToNot(BeNil())
+				Expect(*cdi.Spec.Config.LogVerbosity).To(Equal(logVerbosity))
+			})
+
+			DescribeTable("Should not be defined for KubevirtCR if not defined in HCO CR", func(logConfig *hcov1beta1.LogVerbosityConfiguration) {
```
Comment (nunnatsa) [inline controllers/operands/cdi_test.go:530]: KubevirtCR ==> CDI CR

File: docs/cluster-configuration.md [hunk 3]
```diff
@@ -897,6 +897,21 @@ spec:
...
+For example:
+```yaml
+kind: HyperConverged
+metadata:
+  name: kubevirt-hyperconverged
+spec:
+  logVerbosityConfig:
+    cdi:
+      logVerbosity: 4
+```
```
Comment (nunnatsa) [inline docs/cluster-configuration.md:913]: According to the new API that defined above, there is no nested field `logVerbosity` within the `cdi` field.
I think it should be
```suggestion
```yaml
kind: HyperConverged
metadata:
  name: kubevirt-hyperconverged
spec:
  logVerbosityConfig:
    cdi: 4
```
```

File: docs/cluster-configuration.md [hunk 4]
```diff
@@ -897,6 +897,21 @@ spec:
 All the values defined [here](https://kubevirt.io/api-reference/master/definitions.html#_v1_logverbosity)
 can be applied.
 
+### Containerized data importer (CDI)
+Different levels of verbosity are used in CDI to control the amount of information that is logged. The verbosity level of logs in CDI can be adjusted using a dedicated field that affect all of its components. 
+This feature enables users to control the amount of detail displayed in logs, ranging from minimal to detailed debugging information.
```
Comment (nunnatsa) [inline docs/cluster-configuration.md:902]: Can you please add what values are supported? what each value means?

**Review by nunnatsa — COMMENTED** [review]
Thanks for the PR. Added several inline comments.

File: docs/cluster-configuration.md (hunk 3 above)
Comment (alromeros) [reply docs/cluster-configuration.md:913]: Right, thanks for the clarification!

**Comment by nunnatsa** [comment]
Looks good. Some auto generated files are not updated. Please run "make sanity"

File: docs/cluster-configuration.md (hunk 3 above)
Comment (nunnatsa) [reply docs/cluster-configuration.md]: Thanks for fixing. But if the values are 1-3 - please don't use 4 (-:

Comment (alromeros) [reply docs/cluster-configuration.md]: I changed it to 3, the 4 is from the old comment.

**Review by nunnatsa — APPROVED** [review]

**Pull request merged**